src/v/iceberg/values_json.h

```diff
@@ -0,0 +1,21 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "iceberg/datatypes.h"
+#include "iceberg/json_writer.h"
+#include "iceberg/values.h"
+#include "json/document.h"
+
+namespace iceberg {
+std::optional<value>
```

Comment:
docs please. Point to the spec, where it's expected to be used, etc

Reply:
done